edge/cmd/edgecore/app/server.go

```diff
@@ -87,17 +88,28 @@ offering HTTP client capabilities to components of cloud to reach HTTP servers r
 				}
 			}
 
+			ip := net.ParseIP(config.Modules.Edged.NodeIP)
+
 			// Get edge node local ip only when the custiomInterfaceName has been set.
 			// Defaults to the local IP from the default interface by the default config
-			if config.Modules.Edged.CustomInterfaceName != "" {
+			if ip == nil && config.Modules.Edged.CustomInterfaceName != "" {
 				klog.Infof("Getting IP address by custom interface: %s", config.Modules.Edged.CustomInterfaceName)
-				ip, err := netutil.ChooseBindAddressForInterface(config.Modules.Edged.CustomInterfaceName)
+				ip, err = netutil.ChooseBindAddressForInterface(config.Modules.Edged.CustomInterfaceName)
 				if err != nil {
 					klog.Errorf("Failed to get IP address by custom interface: %s: %s", config.Modules.Edged.CustomInterfaceName, err)
 					os.Exit(1)
 				}
 				config.Modules.Edged.NodeIP = ip.String()
 			}
+
+			if ip == nil {
+				nodeIP, err := util.GetLocalIP(util.GetHostname())
+				if err != nil {
+					klog.Errorf("Failed to get Local IP address: %s", err)
+					os.Exit(1)
+				}
```

Comment:
function `NewDefaultEdgeCoreConfig` has called util.GetLocalIP to generate node ip, so i think this is duplicate.

Reply:
yes， but the default config will be overrided by config（edgecore.yaml）
case1:  nodeIP="", customInterfaceName unset, node-ip is nil.
case2: nodeIP="abc", customInterfaceName unset, node-ip is nil.

Reply:
If the NodeIP and CustomInterfaceName all have values, which one should we choose? In kubelet it has no `CustomInterfaceName`, so I'm a little hesitant, is this value CustomInterfaceName necessary? Because we can config nodeip.

cc @siredmar

Reply:
@chenchunxiu @fisherxu if both `NodeIP` and `CustomInterfaceName` are set i suggest using the IP from `CustomInterfaceName`. 
Why is that? If the user excplcitly wants to obtain the IP address fom a custom interface, he typically knows exactly what he is doing. Please look at the table below for further elaboration.

So my suggested priorities are like this:
```
| CustomInterfaceName | NodeIP  | Usage                       | Info/Warning/Error                                      |
|---------------------|---------|-----------------------------|---------------------------------------------------------|
| valid               | valid   | CustomInterfaceName         | Info: using CustomInterfaceName <value> with IP <value> |
| valid               | invalid | CustomInterfaceName         | Info: using CustomInterfaceName <value> with IP <value> |
| valid               | unset   | CustomInterfaceName         | Info: using CustomInterfaceName <value> with IP <value> |
| invalid             | valid   | None -> Error out           | Error: invalid CustomInterfaceName specified: <value>   |
| invalid             | invalid | None -> Error out           | Error: invalid CustomInterfaceName specified: <value>   |
| invalid             | unset   | None -> Error out           | Error: invalid CustomInterfaceName specified: <value>   |
| unset               | valid   | NodeIP from config          | Info: using IP address <value>                          |
| unset               | invalid | None -> Error out           | Error: invalid IP address specified: <value>            |
| unset               | unset   | NodeIP from util.GetLocalIP | Info: using IP address  <value>                         |
```

Reply:
@fisherxu the `CustomInterfaceName` was introduced by me to allow the user to specify exactly the interface he needs to use. Maybe we have some WIFI, ethernet and LTE on the device. Which IP shall be the right one? With `CustomInterfaceName` the user can specify the exact connection to work with. So yes, in my opinion this i a pretty important feature.

Reply:
yes，I think use CustomInterfaceName first is better too. So  I adjusted the order according to siredmar‘s suggestion